### PR TITLE
Add Native Code Debugging CheckBox to the Debug prop page

### PR DESCRIPTION
**Pull request by etbyrd**
Fix for https://github.com/dotnet/project-system/issues/1125
Assuming I am reading the issue correctly, this only changes "nativeDebugging" in launchSettings.json for the currently selected profile 

Adds the check box to the bottom:
new_debug
Like non-Core projects:
old_debug

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs [hunk 1]
```diff
@@ -296,6 +296,38 @@ public bool HasLaunchOption
...
+            get
+            {
+                if (!IsProfileSelected)
+                {
+                    return false;
+                }
+
+                if (SelectedDebugProfile.OtherSettings.Keys.Contains("nativeDebugging"))
+                {
+                    return (bool) SelectedDebugProfile.OtherSettings["nativeDebugging"];
```
Comment (davidwengier) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs]: Nit: I don't think we have spaces after casts.. could be missing from .editorconfig?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/PropertyPageResources.resx [hunk 2]
```diff
@@ -246,4 +246,7 @@
   <data name="OKBtnHelpText" xml:space="preserve">
     <value>Create the profile and return to the debug property page with the new profile selected</value>
   </data>
+  <data name="chkNativeCodeDebuggingText" xml:space="preserve">
+    <value>Enable native code debugging</value>
```
Comment (davidwengier) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/PropertyPageResources.resx]: Accelerator key?

Though now that I think about it, can we reuse the same resource from non-Core? Then the translations will be the same? Or are those resources still in the other DLL in the VS tree?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs [hunk 3]
```diff
@@ -296,6 +296,38 @@ public bool HasLaunchOption
...
+        public bool NativeCodeDebugging
+        {
+            get
+            {
+                if (!IsProfileSelected)
+                {
+                    return false;
+                }
+
+                if (SelectedDebugProfile.OtherSettings.Keys.Contains("nativeDebugging"))
```
Comment (davidwengier) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs]: ContainsKey would be better:
```suggestion
                if (SelectedDebugProfile.OtherSettings.ContainsKey("nativeDebugging"))
```

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/PropertyPageResources.resx (hunk 2 above)
Comment (etbyrd) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/PropertyPageResources.resx]: I asked Tom about this and he said not to worry about it - the loc folks should have a database to pull from

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs (hunk 3 above)
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs]: Use TryGetValue

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs [hunk 4]
```diff
@@ -296,6 +296,38 @@ public bool HasLaunchOption
...
+            }
+            set
+            {
+                //Unlike other properties that have default values, nativeDebugging may not be set yet. Because false is the default behavior adding it has no affect
+                if (!SelectedDebugProfile.OtherSettings.ContainsKey("nativeDebugging"))
+                {
+                    SelectedDebugProfile.OtherSettings.Add("nativeDebugging", false);
+                }
+
+                if (SelectedDebugProfile != null && (bool) SelectedDebugProfile.OtherSettings["nativeDebugging"] != value)
```
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs]: It seems simpler if you just write the following, also avoids "dirtying" the in memory model when the value is set to false and it's not present in the settings:

``` C#
            SelectedDebugProfile.OtherSettings.TryGetValue("nativeDebugging", out object previousValue);

            if ((bool)previousValue != value)
            {
                SelectedDebugProfile.OtherSettings["nativeDebugging"] = value;
                OnPropertyChanged(nameof(NativeCodeDebugging));
            }
```

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/PropertyPageResources.resx (hunk 2 above)
Comment (etbyrd) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/PropertyPageResources.resx]: As for the accelerator keys - nothing on the page has them so I will make a separate PR to add those

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/PublicAPI.Unshipped.txt [hunk 5]
```diff
@@ -6,4 +6,5 @@ static Microsoft.VisualStudio.ProjectSystem.VS.PropertyPages.PropertyPageResourc
 static Microsoft.VisualStudio.ProjectSystem.VS.PropertyPages.PropertyPageResources.DeleteBtnHelpText.get -> string
 static Microsoft.VisualStudio.ProjectSystem.VS.PropertyPages.PropertyPageResources.NewBtnHelpText.get -> string
 static Microsoft.VisualStudio.ProjectSystem.VS.PropertyPages.PropertyPageResources.OKBtnHelpText.get -> string
-static Microsoft.VisualStudio.ProjectSystem.VS.PropertyPages.PropertyPageResources.chkLaunchBrowserHelpText.get -> string
\ No newline at end of file
+static Microsoft.VisualStudio.ProjectSystem.VS.PropertyPages.PropertyPageResources.chkLaunchBrowserHelpText.get -> string
+static Microsoft.VisualStudio.ProjectSystem.VS.PropertyPages.PropertyPageResources.chkNativeCodeDebuggingText.get -> string
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/PublicAPI.Unshipped.txt]: Not specific to this PR but it seems weird to have the UI elements as public fields.

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/PublicAPI.Unshipped.txt]: Agreed, was looking at this earlier - why are these public API?

Comment (drewnoakes) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/PublicAPI.Unshipped.txt]: Never mind. Missed that this was a resource.

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/PublicAPI.Unshipped.txt]: But why is this public?

Comment (etbyrd) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/PublicAPI.Unshipped.txt]: Not sure, but it is generated as Public when you edit the resx

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs (hunk 4 above)
Comment (etbyrd) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs]: Good point - changed

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/PublicAPI.Unshipped.txt (hunk 5 above)
Comment (drewnoakes) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/PublicAPI.Unshipped.txt]: Is making these internal out of the question?

Comment (etbyrd) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/PublicAPI.Unshipped.txt]: I don't believe it is out of the question - I will make a PR for this and we can discuss there

**Comment by etbyrd** [comment]
@davkean @davidwengier Can I get another review pass with the new commits?

**Pull request merged**